### PR TITLE
SDK: Make SDK Gutenberg more generic

### PR DESCRIPTION
**Pull request by sirreal**
The SDK Gutenberg tool had become overly specific to build the Jetpack
Gutenberg extensions. Said extensions no longer depend on the SDK tool,
instead relying on their own specific build tool.

Should help remove a Webpack plugin only required for building Jetpack blocks (#30904)

## Testing instructions

SDK builds should still work as expected for the only remaining preset:

`npm run sdk gutenberg client/gutenberg/extensions/presets/o2`

### CODE REVIEW

**Review by jsnajdr — APPROVED** [review]
Thank you! The `gutenberg.js` script is much simpler now and I verified that it still builds the 02 preset correctly.

**Comment by sirreal** [comment]
I'll hold off landing until I get approval from @Automattic/gutenpack and/or @Automattic/calypso-sdk.

**Review by ockham — APPROVED** [review]
:heart: it!

In the long run, we could consider turning these scripts into webpack configs (which is kinda what they are anyway), much like we did for Jetpack Blocks, and rely on Webpack's CLI to run them (which gives us things like a configurable output dir for free).

**Pull request merged**
